### PR TITLE
Add setup and norm checker.

### PR DESCRIPTION
**Pull request by aerorahul**
This PR adds:
- `setup.py` for install
- github action to do norm checking with `pycodestyle`

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    package_dir={'': 'src'},
+    packages=find_packages(where='src'),
+    package_data={
+        '': ['cfg/*'],
+    },
+    include_package_data=True,
+    scripts=scripts,
+    python_requires='>= 3.6',
+    install_requires=[
+        'pyyaml',
```
Comment (CoryMartin-NOAA) [inline setup.py:24]: We will need to change this soon, but fine for now?

Comment (aerorahul) [reply setup.py:24]: what specifically will w need to change? `install_requires`?  I am leaning towards creating a `setup.cfg` file to include metadata so we don't have to touch `setup.py`

Comment (CoryMartin-NOAA) [reply setup.py:24]: will need cartopy, matplotlib, yada yada

Comment (aerorahul) [reply setup.py:24]: yeah sure.  As we add dependencies, we should update that list here.

**Review by CoryMartin-NOAA — APPROVED** [review]

**Pull request merged**